Feature refactor enter node

gnat42 opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT


## Description
The code here is difficult to read and follow. It does multiple duplicate tests/if statements instead of one and sub test for the variants. This re-organizes the code to make fewer un-necessary tests, add comments and be a little easier to follow.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog

Nyholm commented:
This sure is a massive function that definitely need some refactoring. What do you think about try to split this up to smaller and more "easy to read" functions?

gnat42 commented:
So, I had that thought as I was making the little changes I did. I don't feel confident enough that the tests truly cover the situations properly (as evidenced by the other PR I submitted). I guess the tests are pretty good in this area. I'm just still learning the internals of the bundle so am hesitant to make big sweeping changes. Thoughts?

Nyholm commented:
Hm. As a rule of thumb one should not refactor code one does not fully understand. One should also know what tests that will be effected and why. 

I would like the idea of refactoring this method to make sure it gets less complex and easy to understand. But I am not sure this change bring us this win.

gnat42 commented:
Please review the latest refactoring. Its much simpler now.

Nyholm commented on Translation/Extractor/File/FormExtractor.php:
```diff
@@ -109,86 +109,154 @@ public function enterNode(Node $node)
+    /**
+     * @param Node $item
+     * @param $domain
+     * @return bool
```
This function needs documentation. I do not understand what the return value represent. If you parse an empty value node, what does it mean that you return true?

Nyholm commented on Translation/Extractor/File/FormExtractor.php:
```diff
@@ -109,86 +109,154 @@ public function enterNode(Node $node)
+     * @param Node $node
+     * @param $domain
+     * @return bool
+     */
```
Same here, it needs docs. What does this function do? Why? What does the return value mean?

Nyholm commented on Translation/Extractor/File/FormExtractor.php:
```diff
@@ -109,86 +109,154 @@ public function enterNode(Node $node)
+     * @param Node $item
+     * @param $domain
+     * @return bool
+     */
```
Same here, it needs docs. What does this function do? Why? What does the return value mean?

Nyholm commented on Translation/Extractor/File/FormExtractor.php:
```diff
@@ -109,86 +109,154 @@ public function enterNode(Node $node)
+    /**
+     * @param Node $node
+     * @return null|string
+     */
```
Add docs. "Return the value of `translation_domain` or null"

Nyholm commented:
I think this is great. Really big improvement on the readability. I made some comments on a few things though. Also, I found some PSR-2 violations. Please run this class with the php-psr-fixer. 

Since you now are very well familiar with this class (much much more than I am), could you also add some docs for the `parseItem` function?

gnat42 commented:
Hey @Nyholm, I added comments to the functions I added. I haven't actually dug into the parseItem function very much so I doubt I could add anything in great detail there other than to say it is called when we have detected a translatable configuration/string. Not sure if that's that detailed enough to warrant inclusion/addition.

Nyholm commented:
Thank you!
